### ISSUE TITLE
Dingz in  HA Core

### ISSUE THREAD

**Issue by retoo** [siku2/hass-dingz#8]
Hi @siku2 @pail23 

I'm working on and off on dingz component, very similar to yours. Unfortunately yours wasn't available when I started and I have discovered yours just recently :(. 

Until dingz added zero conf discovery, the discovery mechanism wasn't really suitable for Home Assistant core. But now with zero conf I think it should be possible to get dingz support into home assistant core. 

What do you think? What were your reasons to build an HCAS component and not a core component?     

Are you interested in merging our two versions and attempt an inclusion in HA Core? The main difference I see is, that I use python-dingz[1] and now support zero conf. I think you guys have a much better light / dimmer support than my version. 

You can find my current version on a wip branch here: https://github.com/home-assistant/core/compare/dev...retoo:dingz-wip?expand=1

Let me know what you think. I'm situated in Zürich, (I guess you guys are also living in switzerland), if you like a can also do a short call about the topic.

Cheers
Reto

[1] https://pypi.org/project/python-dingz/

**Comment by pail23** [comment]
Hi @retoo
I would appreciate a dingz component in HA core. Why don't you go ahead and create a PR with your code in HA core? I would be happy to help e.g. by adding the dimmer support. What do you think?

**Comment by siku2** [comment]
> What do you think? What were your reasons to build an HCAS component and not a core component?

The main reason was really just that I didn't want to develop and maintain a stand-alone Python library for interacting with the dingz API.

> Are you interested in merging our two versions and attempt an inclusion in HA Core? The main difference I see is, that I use python-dingz[1] and now support zero conf. I think you guys have a much better light / dimmer support than my version.

Absolutely, just let me know if and how I can help out.

**Comment by siku2** [comment]
@retoo, did anything ever come of this? It would be awesome to have Dingz be part of Home Assistant directly.

On a first glance, it doesn't look like python-dingz is still up-to-date to me.

**Comment by retoo** [comment]
@siku2 unfortunately I no longer operate any Dingz devices. I moved and have left my Dingz to the new renter. 

I think I have merged it and fixed all the issues about a year ago. But I wasn't too fluent in the home assistant core design to get the module to point to start a pull request.  

@fabaff is maintaining the upstream project (https://github.com/home-assistant-ecosystem/python-dingz) and was always helpful for changes to python-dingz. 

And another thing, I think dingz now supports better discovery protocols (SSDP I think) and probably can better be integrated. The proprietary discovery was ugly to integrate into home assistant. And they support now mqtt which might open other possibilities.

I hope somebody could pick up the topic, be it based on my or somebody else's work.
